### ISSUE TITLE
Migration to organization

### ISSUE THREAD

**Issue by Neustradamus** [webmin/webmin#876]
It is possible to migrate the personal account to organization?
https://help.github.com/articles/converting-a-user-into-an-organization/

**Comment by Neustradamus** [comment]
@webmin Any news?

**Comment by rostovtsev** [comment]
Yes, it's possible.

Follow the guidelines from the link above! 🙂

**Comment by rostovtsev** [comment]
I will close this issue, as it doesn't seem to be related.

**Comment by swelljoe** [comment]
I think Neustrademus is suggesting Webmin be converted to an organization. I'm not sure what benefits that would provide, though? Webmin doesn't have a lot of contributors, and Jamie prefers submissions (even from Virtualmin folks) to come in via PR, anyway.

**Comment by Neustradamus** [comment]
@iliaross: Can you reopen this ticket?
@swelljoe: Yes the personal account to webmin-account for example, in the same time a webmin organization created and a transfer of projects to the new organization (not a fork).

In more @webmin, same for @virtualmin too.
You will be perfect ^^

**Comment by Neustradamus** [comment]
@iliaross, @swelljoe, @webmin, @virtualmin: Any news guys?

**Comment by rostovtsev** [comment]
@Neustradamus What would that change? Why are you asking for it?

**Comment by Neustradamus** [comment]
@iliaross, @swelljoe, @webmin, @virtualmin: Because it will be better (you are a team) and it is free, example ExifTool has migrated: https://github.com/exiftool

You can see for examples:
- http://github.com/gnome
- http://github.com/kde

**Comment by Neustradamus** [comment]
@iliaross, @swelljoe, @webmin, @virtualmin: Any news?

**Comment by swelljoe** [comment]
We already welcome PRs, I don't think it'll change anything about the workflow for contributors. We're both pretty swamped right now, so I think it's just gonna have to sit on the back burner for a while until we have time to work on it.

**Comment by Neustradamus** [comment]
It will be more professional ^^
And you could also group all softs in only one FREE organization too.
-> https://help.github.com/en/articles/differences-between-user-and-organization-accounts

**Comment by chris001** [comment]
I agree with @Neustradamus it'd be better to group the webmin project and the virtualmin project under one organization (whatever the company or organization is called) and add the girhub usernames to belong to the organization ( @jcameron @swelljoe @iliaross etc..).  It makes more sense vs. having two seemingly unrelated unconnected github projects (and redundant usernames webmin for jamie and virtualmin for joe).

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Any news guys after 1,5 years?
All projects in one organization?

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Any news guys after 1,5 years?
All projects in one organization?

Can you re-open this ticket?
It is not done!
I must to open a new?

Linked to:
- https://github.com/virtualmin/virtualmin-gpl/issues/97

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Can you reopen this ticket?
- It is not done and it is not good to realize this ;)

All projects in one organization?

I must to open a new?

Linked to:
- https://github.com/virtualmin/virtualmin-gpl/issues/97

**Comment by chris001** [comment]
Is it two separate companies?  Webmin + Usermin owned and operated by Jamie, and Virtualmin owned and operated by Joe?  Maybe this is why they're unsure about making one github org for all the server management software?

**Comment by rostovtsev** [comment]
Migration to organisation will not change absolutely anything. Isn't it?

**Comment by chris001** [comment]
* One organization probably would unite/centralize the open source crowd fundraising/soliciting donations efforts, which is a big positive.  
* Expose all potential new users, who discover the software on github, to the full portfolio of apps *Min apps, within the first minute, for the first time.

**Comment by rostovtsev** [comment]
That would be good but that means, that Virtualmin and Cloudmin has to be moved under Webmin username, just like Usermin.

As far as I remember, redirects are  automatically set by GitHub for moved repos? If so, it's good.

**Comment by Neustradamus** [comment]
1/ Personal account to organization: no lost / and you can add admin rights to real personal accounts
Example: https://github.com/exiftool/exiftool/issues/1

2/ Move repository to another place: no lost = redirection old to new
Example: https://github.com/tigaseinc/tigase-server

In the old "space": edit description to inform people where is the new host.

**Comment by rostovtsev** [comment]
All of this is available for personal accounts, including redirects and assigning collaborators (in less powerful way, than for organizations though).

There must be something else? 🙂

**Comment by Neustradamus** [comment]
Time to do it, a little effort ;)

**Comment by rostovtsev** [comment]
I don't mind, personally.

**Comment by Neustradamus** [comment]
@iliaross: Why? Why have several personal accounts for one project base?

**Comment by rostovtsev** [comment]
_Authentic Theme_ was moved under organization a while ago. 

Moving _Webmin/Virtualmin_ requires more time and effort.

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Can you reopen this ticket? It is closed but not done ;)

**Comment by chris001** [comment]
I agree on establishing a github organization, under which you would have each project, virtualmin gpl, webmin gpl, cloudmin gpl, seems to be more suitable, than having one dissociated "personal" repo per gpl project.

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Do you need help? I can do it in 5 minutes after access/rights to all accounts/org :)

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Any news?

It will be done before Christmas 2019?

**Comment by rostovtsev** [comment]
I just had a discussion with Jamie about this, and, right now, it doesn't seem valuable enough for us to invest time into it, as besides manipulations on GitHub, it would also trigger many other things to be changed and tested .

Sorry!

**Comment by chris001** [comment]
> I just had a discussion with Jamie about this, and, right now, it doesn't seem valuable enough for us to invest time into it, as besides manipulations on GitHub, it would also trigger many other things to be changed and tested .

Just asking: Which other things to be changed and tested?

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Changed and tested what?

I do not understand.

All changes in only one organization, no lost, links are here...
2 examples on million migrations/changes:
- https://github.com/nerzhul/ocsms
- https://github.com/tigaseinc/tigase-server

**Comment by rostovtsev** [comment]
> I do not understand.

What we also can not understand is your personal concern about this _negligible_ ~_issue_~!? 

> Which other things to be changed and tested?

Automation and build scripts, I can't tell exactly, but Jamie was just pretty clear about the fact that he is not willing to invest his time into something, that will have zero effect on our end-users.

When I moved _Authentic Theme_ to organization (and only because I wanted to update username (URL)), nothin changed really.

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: But there are no lost of links... There are redirections.
You can test old links of "Authentic Theme"!

It will be solved several points:
- Several personal accounts (virtualmin/webmin/usermin)
- Several places (usermin is already empty)

To a free organization with all products.

First possibilty:

1/ Move, and have:
- https://github.com/virtualmin/webmin
- https://github.com/virtualmin/usermin

2/ Look old links are here:
- https://github.com/webmin/webmin -> https://github.com/virtualmin/webmin
- https://github.com/usermin/usermin -> https://github.com/virtualmin/usermin

3/ Convert virtualmin to organization.

4/ If you want, you can keep webmin and usermin in personal accounts (empty accounts) or migrate to organization.

5/ In old accounts (or organizations), please inform the World that it is needed to go @virtualmin - https://github.com/virtualmin.

Second possibility:

- Move all repositories from virtualmin to usermin (empty by default)
- Rename your virtualmin (now empty) to another name
- Change the usermin name account to virtualmin
- Convert virtualmin to organization
- Move webmin/webmin and usermin/usermin to the virtualmin organization

All it works with redirections and no lost for a perfect world.

**Comment by Neustradamus** [comment]
@jcameron, @iliaross, @swelljoe, @webmin, @usermin, @virtualmin: Any news?

End of 2019 soon!

Can you reopen this ticket or I must open a new?

**Comment by Neustradamus** [comment]
Dear all team members,

I do a little comeback to you after my initial requests done:
- 2018-03-19
- 2019-08-19
- 2020-01-02

Our last discussion has been done 2020-04-25.

So we are now in 2025, 5 years and 7 months after last discussion (or 7 years and 8 months since beginning), it is possible to convert usermin and webmin bad personal accounts like you have done with success with virtualmin?

Your projects can not have a personal account but an organization and it is free.

You must to respect GitHub TOS:
- https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#3-account-requirements

You have two possibilities:
- Converting
- Rename / Create an organization / Transfer repositories (no lost, redirections are here)

Important to know, there is no lost, and GitHub adds redirections after a transfer, for example:
- https://github.com/libreswan has been converted
- https://github.com/exiftool has been converted
- https://github.com/redorescue has been converted
- https://github.com/nerzhul/ocsms -> https://github.com/nextcloud/ocsms
- https://github.com/tigaseinc/tigase-server -> https://github.com/tigase/tigase-server
- https://github.com/boothj5/profanity -> https://github.com/profanity-im/profanity
- https://github.com/asciimoo/searx -> https://github.com/searx/searx

Thanks in advance.

----------------------

Converting:
- https://help.github.com/articles/converting-a-user-into-an-organization/

It is possible to rename this account and create the organization and move/transfer the repository to the organization.

Rename account:
- https://docs.github.com/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-user-account-settings/changing-your-github-username

Create an organization:
- https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/creating-a-new-organization-from-scratch

Transfer a repository:
- https://help.github.com/en/github/administering-a-repository/transferring-a-repository

Like:
- https://github.com/virtualmin (your GOOD organization) <3
- http://github.com/gnome
- http://github.com/kde
- ...

**Comment by Neustradamus** [comment]
Dear @webmin team, @usermin team, @virtualmin team, @jcameron, @iliaross, @swelljoe,

We are in 2026, it is possible to look it?

Recently, today, I have seen that several unsolved tickets have been closed, so it is the moment to recall this important closed one which is unsolved too.

After several years since the beginning (2018-03-19), more 8 years, it is possible to solve it?

You have already done ONE GOOD CHANGE with @virtualmin username which is now a GOOD organization, can you do same with @usermin and @webmin username too?

All details (with legal part) have been added in my previous comment.

This initial request has been done to respect all rules and to have all good features.

Thanks in advance.

**Comment by iliaross** [comment]
> You have already done ONE GOOD CHANGE with [@virtualmin](https://github.com/virtualmin) username which is now a GOOD organization, can you do same with [@usermin](https://github.com/usermin) and [@webmin](https://github.com/webmin) username too?
>> All details (with legal part) have been added in my previous comment.

Can you summarize this and explain why it’s important from a user’s perspective to convert to an organization for a `webmin` account?

**Comment by Neustradamus** [comment]
@iliaross: Sorry for the delay, I have not seen your answer!

About current usermin and webmin bad personal accounts, first, it will respect the GitHub rules.
And your company and you will manage directly like good virtualmin organization.

A personal account is for a person not a project, a company, ...

**Comment by iliaross** [comment]
> A personal account is for a person not a project, a company

I agree with that, but for some reason, Jamie doesn't want to do it. There's nothing I can do about it. You should talk to him...

**Comment by Neustradamus** [comment]
@iliaross: Have you tried to explain to @jcameron, all the different points?

Recall, it takes few seconds to do it.

Note: All the @virtualmin team members can modify (like you have already done) the visibility from "Private" to "Public" here too:
- https://github.com/orgs/virtualmin/people

It is important for SEO, visibility too.

**Comment by iliaross** [comment]
I would personally prefer to have all our repos under the `webmin` account. It would make it easier to organize things, like moving issues that users often create in the wrong place, managing secrets, and some other benefits.

But Jamie and Joe aren’t too excited about it. And it’s silly to think they don’t already know the pros and cons of making it an organization and need further explanation. They probably have other reasons or just don’t have the time or confidence that it would bring any real benefits.

In fact, for users, it doesn’t change much, and for us, me in particular—we can handle it either way.

> * https://github.com/orgs/virtualmin/people

I don’t know why `webmin` user isn't hidden and appears in “people” section. 🙂

**Comment by Neustradamus** [comment]
@iliaross: You have not understood me.

It is to convert this personal account to an orgnization like current virtualmin.
All good features are already virtualmin org.

**Comment by iliaross** [comment]
I'm sorry, but I think you're the one who didn't understand. I was trying.

**Comment by jcameron** [comment]
So I finally have time to look into this some more. 

We do have a `webmin` account this is separate from our personal accounts, which I assume we could just convert by following https://help.github.com/articles/converting-a-user-into-an-organization/ . @iliaross are there *any* downsides to this?

**Comment by Neustradamus** [comment]
@jcameron: Thanks for your comment, I really appreciate! :)

Yes, it is the initial request, in the past, the virtualmin account has been converted with success :)

Currently, there are always: webmin and usermin accounts.

Recall:
- A project/company must have an organization.
- You will manage directly with your good personal account like virtualmin organization.

When you modify your default status from private to public, the organization is showed in your profile at left side and when you create commits, the organization is showed too in the list. Can you modify your visibility?
- https://github.com/orgs/virtualmin/people

You can see the @iliaross profile for example.

Of course, the same step must be done for webmin and usermin when it will be done.
You are important in these projects.

**Comment by iliaross** [comment]
> [@iliaross](https://github.com/iliaross) are there _any_ downsides to this?

I will let you know a bit later tomorrow.

**Comment by iliaross** [comment]
> > [@iliaross](https://github.com/iliaross) are there _any_ downsides to this?
> 
> I will let you know a bit later tomorrow.

We're still discussing it internally.

**Comment by Neustradamus** [comment]
@iliaross: Always on the subject?

**Comment by Neustradamus** [comment]
@iliaross: What is your progress about it?

**Comment by Neustradamus** [comment]
@iliaross: What is your progress about it?

**Comment by iliaross** [comment]
I’m not the one making progress on this—Jamie is.

**Comment by Neustradamus** [comment]
@iliaross: Thanks for your answer!

I had not since your "We're still discussing it internally.".

I hope that @jcameron can look for webmin and usermin personal accounts.

Recall: virtualmin is already perfect:
- https://github.com/virtualmin

**Comment by Neustradamus** [comment]
@jcameron: Have you progressed on it?

Thanks in advance.

**Comment by jcameron** [comment]
No I havne't done anything on this due to the potential riskyness of the current migration processs.
\

**Comment by Neustradamus** [comment]
@jcameron: What is the potential riskyness?

No risk: Start by renaming "usermin" account to "usermin-temp-before-removal".
Create the "usermin" organization and add ownership to your good personal account.
Connect with your good personal account, validate the ownership (not only member).
Connect now with to the "usermin-temp-before-removal" account and delete it here: https://github.com/settings/admin.
Easy, the account is empty.

Rename with the same strategy "webmin" account to "webmin-temp".
Create the "webmin" organization and add ownership to your good personal account.
Connect with your good personal account, validate the ownership (not only member).
Connect now with to the "webmin-temp" account, time to start the transfer to the organization (before transfer, no lost all redirections are here), after all transfers into the organization, it is the moment to delete the bad account.

Finish.

**Comment by iliaross** [comment]
> Create the "usermin" organization and add ownership to your good personal account.

Why creating `usermin` organization?

**Comment by Neustradamus** [comment]
@iliaross: Like I have, always, specified, a long time ago, to replace the bad personal account and to respect the GitHub TOS.
The organization is managed by the good team members (one member or more).

And after, no problem to move "usermin" repository from "webmin" organization to "usermin" organization.
All redirections will be always here.

**Comment by iliaross** [comment]
I think it would make sense if we could all agree to move all our repos under the new `webmin` organization, including the Virtualmin and Cloudmin ones. But we don't seem to be able to reach a consensus on this.

And, at this point, having `webmin` as an organization gives little benefit to anyone, as in practice only Jamie and I are working on it.

**Comment by Neustradamus** [comment]
@iliaross: There is no problem to have different organizations :)
A place for everything...

Currently, virtualmin is a good organization (like I have requested in the past).

usermin must have, impossible to have a personal account for a project, empty or not empty (with the usermin repository).
cloudmin must have, impossible to have a personal account for a project, empty or not empty (with the cloudmin repositories).
webmin must have, impossible to have a personal account for a project.

About usermin repository: It is possible to move temporary (or long term) -> to create the redirection between 2 organizations.
https://github.com/usermin/usermin <-> https://github.com/webmin/usermin

I recall: No lost, redirections are here.

Easy to start with usermin and cloudmin first, 2 empty places.

**Comment by iliaross** [comment]
> There is no problem to have different organizations

It is no problem, of course, and we do currently. Yet, it makes little sense to me in terms of hierarchy. Neither Virtualmin nor Cloudmin can exist as separate projects. They are Webmin modules.

> Easy to start with usermin and cloudmin first, 2 empty places.

I hear you, but you need to hear and respect what @swelljoe and @jcameron currently think about this problem. Perhaps that will change in the future, but for now, we should likely put this off for some time. Sorry.

**Comment by Neustradamus** [comment]
@iliaross: User accounts are the problem, it can not be possible.
Converting has not done several years ago ;)
Now, it is needed to do different steps.

Can you look when I have created the ticket? ;)
I have a respect to the team. I always try to help you in communication and to respect GitHub TOS.

**Comment by iliaross** [comment]
> I have a respect to the team. I always try to help you in communication and to respect GitHub TOS.

Thanks. Just to clarify, we don’t share the `webmin` account. Everyone uses their own GitHub account with separate permissions. So this isn’t a TOS issue.

Since GitHub has deprecated direct account conversion, we could contact GitHub Support and ask them to coordinate the migration and protect the `webmin` name, in theory it should be possible.

But, let me repeat myself, I personally see value in keeping all our projects under one organization, like `webmin`. It would let us move issues created in the wrong repository, manage everything in one place, and give Webmin, Usermin, Virtualmin, and Cloudmin one clear home, similar to GNOME (324 repositories), KDE (1.2k repositories), or Nextcloud (372 repositories) and etc, which is a common and good practice.

Stars would remain separate for each repository, but all projects would be presented together under one recognizable organization. I think that would be better than creating a separate organization for every project. 

Nevertheless, we still don’t have a consensus. From my perspective, converting only `webmin` while leaving the other repositories separate would bring **little practical benefit**, since it wouldn’t solve any of the points above. It only makes sense if we agree to move everything under one `webmin` organization.

**Comment by Neustradamus** [comment]
@iliaross: I have understood your choice to prefer all in one organization.
But you need other organizations.
I have sent you all steps to do it, and to start by empty place:
- usermin
- cloudmin

And to finish by:
- webmin

You can open two tabs in your browser
- one to rename the current account (before to delete after all steps)
- one to create a new good org
